doc/application/application.rst

```diff
@@ -369,6 +369,75 @@ again.
 
 .. _application_debugging:
 
+
+Custom Board Definition
+***********************
+
+In cases where the board or platform you are developing for is not yet supported
+by Zephyr you can add the board definition to your application and build for
+this board without having to add it to the Zephyr tree.
+
+The structure needed to support an out-of-tree board is similar to how boards
+are maintained in Zephyr making it easy to upstream a board to Zephyr once the
+initial development is done out of Zephyr.
+
+Add the custom board to your application using the following structure:
+
+.. code-block:: console
+
+   boards/
+   CMakeLists.txt
+   prj.conf
+   README.rst
+   src/
+
+where the boards directory hosts the board you are building for:
+
+.. code-block:: console
+
+   .
+   ├── boards
+   │   └── x86
+   │       └── my_custom_board
+   │           ├── doc
+   │           │   └── img
+   │           └── support
+   └── src
+
+
+Documentation (under doc/) and support files (under support/) are optional, but
+will be needed when submitting to Zephyr.
+
+The contents of `my_custom_board` should follow the same guidelines for any
+Zephyr board meaning it needs to provide the following files::
+
+    my_custom_board_defconfig
+    my_custom_board.dts
+    my_custom_board.yaml
+    board.cmake
+    board.h
+    CMakeLists.txt
+    doc/
+    dts.fixup
+    Kconfig.board
+    Kconfig.defconfig
+    pinmux.c
+    support/
+
+
+Once the board structure is in place, you can build for this board by pointing
+the build system to the new board root::
+
+
+   cmake -DBOARD=<board name> -DBOARD_ROOT=<path to boards> ..
+
+
+This will use the custom board configuration and files and will generate the
```

Comment:
How about:

    This will use your custom board configuration and will generate the
    Zephyr binary into your application directory.